src/redux/epics/download-upload-history-epic.js

```diff
@@ -0,0 +1,20 @@
+import { combineEpics } from "redux-observable";
+import FileSaver from "file-saver";
+
+import downloadUploadHistoryActions from "redux/actions/download-upload-history-actions";
+
+const beginDownloadUploadHistory = (action$, store) => {
+  return action$.ofType(downloadUploadHistoryActions.BEGIN_UPLOAD_HISTORY_DOWNLOAD).map(action => {
+    const { upload: { history } } = store.getState();
+    if(history.length) {
```

Comment:
instead of this you could add
```
.filter(() => {
  const { upload: { history } } = store.getState();
  return !!history.length;
})
```

before `.map(...)`